core/access_tokens.go

```diff
@@ -87,6 +88,15 @@ func (a *API) deleteAccessToken(ctx context.Context, x struct{ ID string }) erro
 	if currentID == x.ID {
 		return errCurrentToken
 	}
-	return a.accessTokens.Delete(ctx, x.ID)
-	// TODO(tessr): delete any associated grants, as well
+	err := a.accessTokens.Delete(ctx, x.ID)
+	if err != nil {
+		return err
+	}
+
+	err = a.revokeGrantsByAccessToken(ctx, x.ID)
+	if err != nil {
+		// well, technically we did delete the access token, so don't return the error
+		log.Printkv(ctx, log.KeyError, "revoking grants for access token %s", x.ID)
+	}
```

Comment:
alright, I'll add a TODO. Do you think it's better to return an error when the token gets deleted from postgres, or to fail to delete the grant from raft? 

In general, I'd lean towards simply failing to delete the grant, except we're making some assumptions about the number of grants we store, so if someone were cycling through access tokens regularly we might cause performance problems.